tests/AbstractFixerTestCase.php

```diff
@@ -0,0 +1,658 @@
+        static::assertFalse($emptyTokens->isChanged());
+
+        static::assertNull(
+            $this->fixer->fix(new SplFileInfo(__FILE__), $tokens),
+            sprintf('Return type for ::fix of "%s" is invalid.', $this->fixer->getName()),
+        );
+    }
+
+    protected function getFixer(): FixerInterface
+    {
+        assert($this->fixer instanceof FixerInterface);
+
+        if ($this->fixer instanceof WhitespacesAwareFixerInterface) {
+            $this->fixer->setWhitespacesConfig(new WhitespacesFixerConfig());
+        }
+
+        return $this->fixer;
+    }
+
+    final protected function doTest(string $expected, ?string $input = null, ?SplFileInfo $file = null): void
+    {
+        if ($expected === $input) {
+            throw new InvalidArgumentException('Input parameter must not be equal to expected parameter.');
+        }
+
+        $file ??= new SplFileInfo(__FILE__);
+        $fileIsSupported = $this->fixer->supports($file);
+
+        if (null !== $input) {
+            self::assertNull($this->lintSource($input));
+
+            Tokens::clearCache();
+            $tokens = Tokens::fromCode($input);
+
+            if ($fileIsSupported) {
+                self::assertTrue($this->fixer->isCandidate($tokens), 'Fixer must be a candidate for input code.');
+                self::assertFalse($tokens->isChanged(), 'Fixer must not touch Tokens on candidate check.');
+                $this->fixer->fix($file, $tokens);
+            }
+
+            $this->assertSame($tokens->generateCode(), $expected);
+
+            self::assertTrue(
+                $tokens->isChanged(),
+                'Tokens collection built on input code must be marked as changed after fixing.',
+            );
+
+            $tokens->clearEmptyTokens();
+
+            self::assertSameSize(
+                $tokens,
+                array_unique(array_map(static fn (Token $token): string => spl_object_hash($token), $tokens->toArray()),
+                ),
+                'Token items inside Tokens collection must be unique.',
+            );
+
+            Tokens::clearCache();
+            $expectedTokens = Tokens::fromCode($expected);
+            $this->assertSameTokens($expectedTokens, $tokens);
+        }
+
+        self::assertNull($this->lintSource($expected));
+
+        Tokens::clearCache();
+        $tokens = Tokens::fromCode($expected);
+
+        if ($fileIsSupported) {
+            $this->fixer->fix($file, $tokens);
+        }
+
+        $this->assertSame($tokens->generateCode(), $expected);
+        self::assertFalse(
+            $tokens->isChanged(),
+            'Tokens collection built on expected code must not be marked as changed after fixing.',
+        );
+    }
+
+    protected function lintSource(string $source): ?string
+    {
+        try {
+            $this->linter->lintSource($source)
+                ->check()
+            ;
+        } catch (Exception $e) {
+            return $e->getMessage() . "\n\nSource:\n{$source}";
```

Comment:
use sprintf, see https://github.com/paysera/php-style-guide?tab=readme-ov-file#double-quotes

Reply:
This is old code. It is fixed anyway.